scripts/download_artifacts.py

```diff
@@ -0,0 +1,151 @@
+    )
+    return parser.parse_args()
+
+def get_possible_artifact_names():
+    """
+    Generates all possible permutations of target artifact logs and
+    removes unsupported targets
+
+    Current Support:
+      Linux: rv32/64 multilib non-multilib
+      Newlib: rv32/64 non-multilib
+      Arch extensions: gc
+    """
+    libc = ["gcc-linux", "gcc-newlib"]
+    arch = ["rv32{}-ilp32d-{}", "rv64{}-lp64d-{}"]
+    multilib = ["multilib", "non-multilib"]
+
+    arch_extensions = ["gc"]
+
+    all_lists = ["-".join([i, j, k]) for i in libc
+                                     for j in arch
+                                     for k in multilib
+                                     if not ("rv32" in j and k == "multilib")]
+
+    all_names = [name.format(ext, "{}") for name in all_lists
+                                        for ext in arch_extensions]
+    return all_names
+
+def check_artifact_exists(artifact_name: str, repo: Repository):
+    """
+    @param artifact_name is the artifact associated with build success
+    If the artifact does not exist, something failed and logs error into
+    appropriate file; else returns artifact id for downloading
+    """
+    build_name = artifact_name
+    build_failed = False
+    artifact = repo.get_artifacts(artifact_name).get_page(0)
+    # check build failed
+    if len(artifact) == 0:
+        print(f"build failed for {build_name}")
+        build_failed = True
+        with open("./logs/failed_build.txt", "a+") as f:
+            f.write(f"{build_name}|Check logs\n")
+    # check testsuite failed
+    artifact_name += "-report.log"
+    artifact = repo.get_artifacts(artifact_name).get_page(0)
+    if len(artifact) == 0:
+        print(f"testsuite failed for {build_name}")
+        if not build_failed:
+            with open("./logs/failed_testsuite.txt", "a+") as f:
+                f.write(f"{build_name}|Cannot find testsuite artifact\n")
+        return -1
+    return artifact[0].id
+
+def download_artifact(artifact_name: str, artifact_id: str, token: str):
+    """
+    Uses GitHub api endpoint to download and extract the log artifact into
+    directory called ./logs
+    """
+    params = {"Accept": "application/vnd.github+json",
+              "Authorization": f"token {token}",
+              "X-Github-Api-Version": "2022-11-28"}
+    artifact_zip_name = artifact_name.replace(".log", ".zip")
+    print(artifact_zip_name)
+    print(f"downloading: {artifact_id}")
+    r = requests.get(f"https://api.github.com/repos/patrick-rivos/riscv-gnu-toolchain/actions/artifacts/{artifact_id}/zip", headers=params)
+    print(r.status_code)
+    with open(f"./{artifact_zip_name}", "wb") as f:
+        f.write(r.content)
+    with ZipFile(f"./{artifact_zip_name}", "r") as zf:
+        zf.extractall(path=f"./{artifact_name.split('.log')[0]}")
+    os.rename(f"./{artifact_name.split('.log')[0]}/{artifact_name}", f"./logs/{artifact_name}")
+
+def download_all_artifacts(current_hash: str, token: str):
```

Comment:
If this function both downloads and runs comparisons, we should note that in the function name.